### PR TITLE
Add webhook url to desired chart spec

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/giantswarm#13134

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/app/resource/chart/desired.go [hunk 1]
```diff
@@ -71,6 +72,12 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	annotations := generateAnnotations(cr.GetAnnotations())
+
+	if key.InCluster(cr) {
```
Comment (tomahawk28) [inline service/controller/app/resource/chart/desired.go:76]: Adding webhook annotation only if it's control plane apps.

**Comment by tomahawk28** [comment]
Tested in local kind; 
```
kg get charts userd-unique -o yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: Chart
metadata:
  annotations:
    chart-operator.giantswarm.io/values-md5-checksum: 2eb49550924800375d501c8de34e0656
    chart-operator.giantswarm.io/webhook-url: http:/app-operator-unique.giantswarm:8000/status/giantswarm/userd-unique
  creationTimestamp: "2020-10-26T15:28:41Z"
  finalizers:
  - operatorkit.giantswarm.io/chart-operator-chart
```

**Comment by tomahawk28** [comment]
Now it is fixed as below ```kg get charts userd-unique -o yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: Chart
metadata:
  annotations:
    chart-operator.giantswarm.io/values-md5-checksum: 2eb49550924800375d501c8de34e0656
    chart-operator.giantswarm.io/webhook-url: http://app-operator-unique.giantswarm:8000/status/giantswarm/userd-unique
  creationTimestamp: "2020-10-26T15:28:41Z"
  finalizers:
  - operatorkit.giantswarm.io/chart-operator-chart```

File: helm/app-operator/templates/configmap.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ data:
     service:
       app:
         unique: {{ include "resource.app.unique" . }}
+      chart:
+        webhookBaseURL: 'http://{{ include "resource.default.name" . }}.{{ include "resource.default.namespace" . }}:{{ .Values.port }}'
```
Comment (tomahawk28) [inline helm/app-operator/templates/configmap.yaml:20]: Build a webhook base URL from flags, and use it from resource-level is more efficient.

**Review by rossf7 — APPROVED** [review]
Nice, LGTM

**Pull request merged**